ethernet: Add 2.5G and 5G Ethernet defines

go2sh opened the pull request:
Add required defines for 2.5G and 5G ethernet standards.

Will be used in conjunction with the xgmac driver.

pdgendt commented on include/zephyr/net/ethernet.h:
```diff
@@ -204,6 +204,12 @@ enum ethernet_hw_caps {
 	ETHERNET_TXINJECTION_MODE	= BIT(20),
+
+	/** 2.5 Gbits link supported */
+	ETHERNET_LINK_2_5GBASE_T	= BIT(21),
+
+	/** 5 Gbits link supported */
+	ETHERNET_LINK_5GBASE_T		= BIT(22),
```
```suggestion
	ETHERNET_LINK_2500BASE_T	= BIT(21),

	/** 5 Gbits link supported */
	ETHERNET_LINK_5000BASE_T	= BIT(22),
```

Seems to be more in line with other types.

carlescufi commented:
Thanks for the PR! Please take a look at [our commit message guidelines](https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-message-guidelines) to find out how to format your commit messages, and at [our contribution workflow](https://docs.zephyrproject.org/latest/contribute/guidelines.html#contribution-workflow) to understand how to update your Pull Request.

jukkar commented on include/zephyr/net/phy.h:
```diff
@@ -31,17 +31,21 @@ extern "C" {
 enum phy_link_speed {
 	/** 10Base-T Half-Duplex */
-	LINK_HALF_10BASE_T		= BIT(0),
+	LINK_HALF_10BASE_T  	= BIT(0),
```
Looks like unnecessary change, why do we need to change the white space here?

go2sh replied:
Removed.

jukkar commented on include/zephyr/net/phy.h:
```diff
@@ -51,7 +55,7 @@ enum phy_link_speed {
  *
  * @return True if link is full duplex, false if not.
  */
-#define PHY_LINK_IS_FULL_DUPLEX(x)	(x & (BIT(1) | BIT(3) | BIT(5)))
+#define PHY_LINK_IS_FULL_DUPLEX(x)	(x & (BIT(1) | BIT(3) | BIT(5) | BIT(6) | BIT(7)))
```
unrelated change, please put to separate commit

go2sh replied:
Linked to the new phy link speeds above.

jukkar commented on subsys/net/lib/shell/iface.c:
```diff
@@ -56,6 +56,10 @@ static struct ethernet_capabilities eth_hw_caps[] = {
 	EC(ETHERNET_HW_FILTERING,         "MAC address filtering"),
 	EC(ETHERNET_DSA_SLAVE_PORT,       "DSA slave port"),
 	EC(ETHERNET_DSA_MASTER_PORT,      "DSA master port"),
+	EC(ETHERNET_TXTIME,               "TXTIME supported"),
+	EC(ETHERNET_TXINJECTION_MODE,     "TX-Injection supported"),
```
Unrelated change, please put to different commit

go2sh replied:
Put in a seperate commit.

jukkar reviewed (commented):
Please add `net: ` prefix to commit subject so `net: ethernet: ...`

go2sh committed:
net: ethernet: Add 2.5GBase-T and 5GBase-T phy speeds

The possible link speeds of ethernet phys are extended by
the 2.5G and 5G constants.

Signed-off-by: Christoph Seitz <christoph.seitz@infineon.com>

go2sh committed:
net: shell: Add missing hw_cap strings.

Two ethernet capabilities were missing. Added them to allow
the 2.5G and 5G strings to appended.

Signed-off-by: Christoph Seitz <christoph.seitz@infineon.com>

go2sh committed:
net: ethernet: Add 2.5G and 5G hardware capabilities.

Add the 2.5G and 5G link speeds as possible capabilities
to ethernet drivers.

Signed-off-by: Christoph Seitz <christoph.seitz@infineon.com>

go2sh commented:
@jukkar @carlescufi Addressed your feedback..

rlubos reviewed (commented):
Please fix the compliance issue:
```
3: UC4 Commit message body line exceeds max length (77>75): "Add the 2.5G and 5G link speeds as possible capabilities to ethernet drivers."
```

jukkar commented:
ping @go2sh

go2sh commented:
@jukkar Done. Sorry for the delay. Some customer topics put away from working on zephyr.
